pipeline/api/cloud.py

```diff
@@ -335,51 +353,36 @@ def _post(self, endpoint, json_data):
 
         return response.json()
 
-    def _post_file(self, endpoint, file, remote_path) -> FileGet:
+    def _post_file(self, endpoint: str, file: io.BytesIO) -> FileGet:
         self.raise_for_invalid_token()
         if not hasattr(file, "name"):
             file.name = generate_id(20)
 
-        e = encoder.MultipartEncoder(
-            fields={
-                "file_path": remote_path,
-                "file": (
-                    file.name,
-                    file,
-                    "application/octet-stream",
-                    {"Content-Transfer-Encoding": "binary"},
-                ),
-            }
-        )
-        encoder_len = e.len
+        file_size = file.getbuffer().nbytes
+
         if self.verbose:
-            bar = tqdm(
+            progress = tqdm(
                 desc=f"{PIPELINE_STR} Uploading",
                 unit="B",
                 unit_scale=True,
-                total=encoder_len,
+                total=file_size,
                 unit_divisor=1024,
             )
-        if self.verbose:
-
-            def progress_callback(monitor):
-                bar.n = monitor.bytes_read
-                bar.refresh()
-                if monitor.bytes_read == encoder_len:
-                    bar.close()
-
-        encoded_stream_data = encoder.MultipartEncoderMonitor(
-            e, callback=progress_callback if self.verbose else None
-        )
+            # If verbose then wrap our file object in a tqdm callback
+            file = CallbackIOWrapper(progress.update, file, "read")
```

Comment:
```suggestion
            file = CallbackIOWrapper(progress.update, file)
```

I guess this code path is lacking coverage.

Reply:
@alexpearce Rather confusingly, we are currently using 2 of these callback wrapper classes: `CallbackIOWrapper` is the one provided by the `tqdm` library, and `CallbackBytesIO` is the one I created (needed to work with `httpx`).

So this code was working fine before but still works now (`"read"` is the default for this argument).

If we want to just use my class for this for simplicity then we could do (we would just need to allow it to accept a `BytesIO` as an input in addition to `bytes`).

Hope that makes sense?

Reply:
Ahh, I missed that there were two classes with the same name!

No strong preference between consolidation or using the two separately; as you like.